aten/src/ATen/Dispatch.h

```diff
@@ -42,5 +42,22 @@ auto dispatch_floating_types(const Type& the_type, const char *name, Args&&... a
     }
 }
 
+#define DISPATCH_ALL_FLOATING_TYPES(TYPE, NAME, F...)                             \
+  [&]() {                                                                         \
+    const Type& the_type = TYPE;                                                  \
```

Comment:
This ensures that if you do something like `DISPATCH_ALL_FLOATING_TYPES(self, ...)` instead of `DISPATCH_ALL_FLOATING_TYPES(self.type(), ...)` you'll get a meaningful compile error early on.

I'm not too worried about another variable in-scope. It can be shadowed in `F` without generating compile warnings.

Reply:
up to you, I don't like people messing with my scope :).  You can get the same behavior by just enclosing the assignment in brackets and using TYPE from then on.